src/StreamedPart.php

```diff
@@ -218,6 +218,11 @@ public function getBody()
                 break;
         }
 
+        // Don't try to fix the encoding when it's a file
+        if ('' === $encoding && $this->isFile()) {
```

Comment:
if the mime type of the part is not text, and the transfer encoding is missing, as per spec it should be considered an error. 

nothing says that if is a file it is binary.


```
Host: localhost:8080
Accept: */*
Content-Type: multipart/form-data; boundary=----------------------------83ff53821b7c

------------------------------83ff53821b7c
Content-Disposition: form-data; name="upload"; filename="text.txt"
Content-Type: text/plain|

XXX
```

the `text.txt` file should be interpreted as 7bit, while your change will interpret it as binary

Reply:
I believe this will not be an issue anymore with the latest changes 🙏